### ISSUE TITLE
Function in Props

### ISSUE THREAD

**Issue by FourwingsY** [storybookjs/addon-jsx#30]
When i put some functions on the Component, the Whole function including `__WEBPACK_IMPORTED_MODULE...` shows on jsx addons.

even with `@storybook/addon-actions`, JSX preview has shown as

```javascript
<Calendar
  onSelect={function action() {
    for (var _len = arguments.length, _args = Array(_len), _key = 0; _key < _len; _key++) {
      _args[_key] = arguments[_key];
    }

    var args = _args.map(_util.prepareArguments);
    var channel = _addons2.default.getChannel();
    var id = (0, _v2.default)();
    channel.emit(_.EVENT_ID, {
      id: id,
      data: { name: name, args: args }
    });
  }}
/>
```

This is not what I expected... but is this intended?

**Comment by ndelangen** [comment]
Hmm, I think we could improve this by adding a toString on the `action` returned function:
https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Function/toString

**Comment by FourwingsY** [comment]
addon-actions can be handled in that way. But in my case, some defaultProps (like customizable renderer props) were printed as well. And it was soo dirty(bundled by webpack).

I will set toString in my functions.

**Comment by apaatsio** [comment]
Here's an example for quick workaround for the `action` function:

```jsx
const onClickAction = action('clicked');
onClickAction.toString = () => "action('clicked')";

storiesOf('Button', module)
  .addWithJSX('default', () => <Button onClick={onClickAction}>Hello</Button>);
```

The output is

```jsx
<Button onClick={action('clicked')}>
  Hello
</Button>
```

**Comment by stramel** [comment]
This is a quick util function I wrote.

```js
import { action as _action } from '@storybook/addon-actions'

export const action = (...args) => {
  const tempAction = _action(...args)
  tempAction.toString = () => `action('${args[0]}')`
  return tempAction
}
```

to use it just import the util function instead of the library function.

```js
import { action } from './utils'
```

**Comment by JasonTheAdams** [comment]
@ndelangen This has been around a bit and it would be handy to have this handled in the Storybook repo. Is there a corresponding issue or PR for this?

**Comment by crimx** [comment]
You can do something like this

```
addParameters({
  jsx: {
    functionValue: (fn: Function) => `${fn.name}()`
  }
})
```

**Comment by keiththomps** [comment]
This issue is pretty rough for `render` props. For this story:

```javascript
  <ModalButton text={text("Button text", "Show Modal")} render={close => (
    <div>
      <h1>Testing w/ Render Props</h1>
      <Button onClick={close}>Close</Button>
    </div>
  )} />
```

The JSX addon presents this:

```javascript
<ModalButton
  render={function render(close) {return react__WEBPACK_IMPORTED_MODULE_0___default.a.createElement("div", null, react__WEBPACK_IMPORTED_MODULE_0___default.a.createElement("h1", null, "Testing w/ Render Props"), react__WEBPACK_IMPORTED_MODULE_0___default.a.createElement(_buttons__WEBPACK_IMPORTED_MODULE_3__["Button"], {onClick: close}, "Close"));}}
  text="Show Modal"
/>
```

Since the render props pattern is fairly common I would think that someone has solved this for their storybooks, but I'm pretty new to storybooks and at a bit of a loss as to where I should start.

**Comment by hipstersmoothie** [comment]
A thing like this would have to be solved upstream via https://github.com/algolia/react-element-to-jsx-string/

**Comment by keiththomps** [comment]
@hipstersmoothie Thanks, I'll see if this is something that they'd be interested in having addressed and try to contribute a solution if they'll take it.

**Comment by hipstersmoothie** [comment]
:rocket: Issue was released in [`v7.3.1`](https://github.com/storybookjs/addon-jsx/releases/tag/v7.3.1) :rocket:
